### PR TITLE
Small fixes Import-DbaXESessionTemplate

### LINKED ISSUE #4923: [Import-DbaXESessionTemplate] Name cannot be specified with multiple files or templates because the Session will already exist.
### Before submitting a bug report:

- [x] Ensure you are able to reproduce it on the latest released version (_we release often_)
- [x] Verified this bug is not already reported in an issue
- [x] Verified errors are not related to permissions
- [ ] Can reproduce in a clean PowerShell session (_clean = `powershell -NoProfile`_)

If two or more templates are imported, they can't get the same name. In this case the warning should be shown.

        if (($Path.Count -gt 1 -or $Template.Count -gt 1) -and (Test-Bound -ParameterName Template)) {
            Stop-Function -Message "Name cannot be specified with multiple files or templates because the Session will already exist."
        }

So the second part should be Test-Bound -ParameterName Name

The function continues anyway when the warning is shown. For me it makes more sense if the function just stops after the warning instead of just doing the job partly.

## Steps to Reproduce

```powershell

Case 1:
Import-DbaXESessionTemplate -SqlInstance . -Template 'Database Health 2016 and Above','Activity Tracking'

Case 2:
Import-DbaXESessionTemplate -SqlInstance . -Template 'Database Health 2016 and Above','Activity Tracking' -Name test

## Expected Behavior

The first case should not show the warning. The second case should show the warning:
"Name cannot be specified with multiple files or templates because the Session will already exist."

## Actual Behavior

The first case shows the warning and the second case not.

## Environmental data

<!-- Provide output of the following two commands -->

 - PowerShell: 

Name                           Value                                                                                                                                                                                                                             
----                           -----                                                                                                                                                                                                                             
PSVersion                      5.1.14393.2636                                                                                                                                                                                                                    
PSEdition                      Desktop                                                                                                                                                                                                                           
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                                                                           
BuildVersion                   10.0.14393.2636                                                                                                                                                                                                                   
CLRVersion                     4.0.30319.42000                                                                                                                                                                                                                   
WSManStackVersion              3.0                                                                                                                                                                                                                               
PSRemotingProtocolVersion      2.3                                                                                                                                                                                                                               
SerializationVersion           1.1.0.1   

 - - SQL Server:

Microsoft SQL Server vNext (CTP2.0) - 15.0.1000.34 (X64)   Sep 18 2018 10:18:13   Copyright (C) 2018 Microsoft Corporation  Developer Edition (64-bit) on Windows Server 2016 Datacenter 10.0 <X64> (Build 14393: ) (Hypervisor)

### PR DESCRIPTION
**Pull request by ijeb**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes #4923)
 - [ ] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
Fix a wrong parameter check.

### Commands to test
Import-DbaXESessionTemplate -SqlInstance . -Template 'Database Health 2016 and Above','Activity Tracking' -Name test -Verbose

### CODE REVIEW

File: functions/Import-DbaXESessionTemplate.ps1 [hunk 1]
```diff
@@ -95,7 +95,7 @@ function Import-DbaXESessionTemplate {
             Stop-Function -Message "You must specify Path or Template."
         }
 
-        if (($Path.Count -gt 1 -or $Template.Count -gt 1) -and (Test-Bound -ParameterName Template)) {
+        if (($Path.Count -gt 1 -or $Template.Count -gt 1) -and (Test-Bound -ParameterName Name)) {
```
Comment (wsmelton) [inline functions/Import-DbaXESessionTemplate.ps1]: Can you clarify why this change is required?

Template is a valid parameter and as documentation states:

> Specifies the name of one of the templates from the dbatools repository

Comment (wsmelton) [reply functions/Import-DbaXESessionTemplate.ps1]: Name is for the name of the XE session to be created.

**Comment by potatoqualitee** [comment]
thanks for the review, shawn! and welcome back @ijeb 💃

File: functions/Import-DbaXESessionTemplate.ps1 (hunk 1 above)
Comment (ijeb) [reply functions/Import-DbaXESessionTemplate.ps1]: I attached issue #4923 to it. This caught my attention while scanning the code trying to understand what is happening.
The warning message contradicts the code,

Comment (wsmelton) [reply functions/Import-DbaXESessionTemplate.ps1]: Based on the message then Template should still be checked and the Name parameter simply added to the logic:
```suggestion
        if (($Path.Count -gt 1 -or $Template.Count -gt 1) -and (Test-Bound -ParameterName Template) -and (Test-Bound -ParameterName Name)) {
```

Comment (ijeb) [reply functions/Import-DbaXESessionTemplate.ps1]: @wsmelton 

The message is: "Name cannot be specified with multiple files or templates because the Session will already exist."

Notice that the parameter -Path also specifies a template, but it needs the full path and the file extention. Apart from this, the parameter -Template has the same function.

Breaking it into pieces:
"Name cannot be specified" : (Test-Bound -ParameterName Name)
"multiple files or templates" : ($Path.Count -gt 1 -or $Template.Count -gt 1)
	--> multiple files means $Path.Count -gt 1, sames goes for templates

For me it's still clear the developer just made a typo, using the wrong parameter. Very small issue.

To prove my case further, let me show this test:

Import-DbaXESessionTemplate -SqlInstance vmSQL -Path 'C:\Users\ijeb\Documents\GitHub\dbatools\bin\XEtemplates\Failing Queries.xml','C:\Users\ijeb\Documents\GitHub\dbatools\bin\XEtemplates\Function Executions.xml' -Name 'NewSession'

This should give the warning and abort, because it is not possible to have two sessions with the same name.

With the code I committed the result is:
WARNING: [08:23:07][Import-DbaXESessionTemplate] Name cannot be specified with multiple files or templates because the Session will already exist.

And with your suggested change (after deleting NewSession manually) the result is:

ComputerName : vmSQL
InstanceName : MSSQLSERVER
SqlInstance  : vmSQL
Name         : NewSession
Status       : Stopped
StartTime    : 
AutoStart    : False
State        : Existing
Targets      : {}
TargetFile   : {}
Events       : {sqlserver.error_reported}
MaxMemory    : 4096
MaxEventSize : 0

WARNING: [08:27:03][Import-DbaXESessionTemplate] NewSession already exists on vmSQL

So no warning is shown. This is wrong.

However, today I thought of another possibility. It is possible to use the parameters -Path and -Template at the same time.

This should also show the warning, but it does not
Import-DbaXESessionTemplate -SqlInstance vmSQL -Path 'C:\Users\ijeb\Documents\GitHub\dbatools\bin\XEtemplates\Failing Queries.xml' -Template 'Function Executions' -Name 'NewSession'

So $Path.Count equals 1
$Template.Count equals 1

These two numbers need to be added together. The developer did not think of this case.

This can be solved

if ((($Path.Count + $Template.Count) -gt 1) -and (Test-Bound -ParameterName Name)) {

However, in this case the warning message is not good anymore. I would suggest:

It is not possible to import multiple templates together and give them the same name. Either import one template at a time or do not use the Name parameter.

What do you think of this?

Comment (wsmelton) [reply functions/Import-DbaXESessionTemplate.ps1]: The intention is not for you to use Template with the Path parameter. Path is only used when you want to provide your own custom template. The Template path is used when you want to utilize the templates we include in the module (the command will iterate through and find the file that matches the Template name). 

So you **can** use:
- Name + Path
`Import-DbaXeSessionTemplate … -Path '<path to my xml file>' -Name MySession`
- Name + Template
`Import-DbaXeSessionTemplate … -Template 'Profiler Tuning' -Name MySession`

Combining the following parameters should throw a message as invalid parameter combinations:
- Path + Template
- Path + Template + Name

Comment (ijeb) [reply functions/Import-DbaXESessionTemplate.ps1]: This is leading nowhere.
My intention is to only change a small inconsistency in the code. I don't want to change the original design.
How should I clean this up in a clean way? Just delete the branch / revoke the PR?

**Comment by potatoqualitee** [comment]
awesome, thanks so much to your both!
